File: youtube-links/src/shared/features/links/commands/CreateLink.ts
import { ValidationConsts } from '@/shared/localization/ValidationConsts';

export namespace CreateLink {
  export interface Command {
    url: string;
    playlistId: number;
  }

  export const Validation = {
    url: {
      notEmpty: (v: string) => !!v || 'Youtube video url should not be empty.',
      matchesYoutubeVideoUrlRegex: (v: string) =>
        ValidationConsts.matchesYoutubeVideoUrlRegex(v) ||
        'This is not a valid link to the YouTube video.',
    },
  };
}

File: YoutubeLinks.Shared/Features/Links/Commands/CreateLink.cs
using FluentValidation;
using MediatR;
using Microsoft.Extensions.Localization;
using YoutubeLinks.Shared.Localization;

namespace YoutubeLinks.Shared.Features.Links.Commands;

public static class CreateLink
{
    public class Command : IRequest<int>
    {
        public string Url { get; set; }
        public int PlaylistId { get; set; }
    }

    public class Validator : AbstractValidator<Command>
    {
        public Validator(IStringLocalizer<ValidationMessage> localizer)
        {
            RuleFor(x => x.Url)
                .NotEmpty()
                .WithMessage(x => localizer[nameof(ValidationMessageString.UrlNotEmpty)])
                .Matches(ValidationConsts.YoutubeVideoUrlRegex)
                .WithMessage(x => localizer[nameof(ValidationMessageString.VideoUrlMatchesRegex)]);
        }
    }
}